### PR TITLE
Add Stacked Bar Chart Pattern

### PR DESCRIPTION
**Pull request by mresti**
## Description
Adds implementation for Stacked Bar Chart.

Address #556 

## Changes

* Adds functions for default settings for Stacked Bar Chart.
* Adds tests pages with Stacked Bar Charts.
* Adds Stacked Bar Charts in Bar Chart specs.

## Link to image

This is an image:

Bar Charts

### CODE REVIEW

**Comment by andresgalante** [comment]
@mresti This is awesome, thanks a lot! 👍

**Comment by jennyhaines** [comment]
@mresti - Love the color update, and the chart looks completely consistent with other Patternfly bar charts. Great Job. The only issue I noticed was that the dataset descriptor ("2014") doesn't show up in the current stacked vertical bar chart. You might want to also check the horizontal one. Thanks- It will be good to go once that is fixed.

**Comment by jennyhaines** [comment]
@mresti - Would you mind providing me a link to the Stacked Bar Charts? I'd like to see them in action and inspect the code for spacing, as I need to spec the design out for PatternFly. Thank you!

**Comment by mresti** [comment]
@jennyhaines @andresgalante thanks!

@jennyhaines Oh yeah!! The link is https://rawgit.com/mresti/patternfly/stacked_bar_chart_dist/dist/tests/bar-charts.html  I had reviewed the code and I hope that the issue is fixed. Can give me the feedback? Thank you!

**Comment by jennyhaines** [comment]
@mresti - Nice fix, so we're good to go now. And thanks for the link!

File: src/js/patternfly-settings.js [hunk 1]
```diff
@@ -200,6 +200,26 @@
...
+          position: 'bottom'
+        };
+      },
+      getDefaultStackedBarConfig = function () {
+        return {
+          color: this.getDefaultColors(),
+          grid: this.getDefaultStackedBarGrid(),
+          legend: this.getDefaultStackedBarLegend()
+        };
+      },
```
Comment (dlabrecq) [inline src/js/patternfly-settings.js:222]: This is more FYI, but we've split up the patternfly-settings.js file, which is about to be merged into a feature branch, named branch-4.0-dev. I believe this code will now live in a new file, named patternfly-settings-charts. Just need to ensure this change is added to the new branch as well.

File: tests/unit/bar-charts.spec.js [hunk 2]
```diff
@@ -85,6 +85,44 @@ describe("bar-charts test suite", function () {
...
+      columns: stackedColumnsData,
+      groups: stackedGroups,
+      // optional drilldown behavior
+      onclick: function (d, element) {
+        window.location = chartUrls[d.index];
+      }
+    };
+    stackedVerticalBarChartConfig.color = stackedColors;
+    var stackedVerticalBarChart = c3.generate(stackedVerticalBarChartConfig);
+
```
Comment (dlabrecq) [inline tests/unit/bar-charts.spec.js:148]: I'd like @priley86 to comment on the unit tests as he implemented the existing test cases. It doesn't appear that we're testing much here?

Patrick, is the idea of this particular unit test to simply render the charts? Do we need more?

**Review by dlabrecq — COMMENTED** [review]
LGTM, but would like @priley86 or @bleathem to comment on the unit tests.

**Comment by jennyhaines** [comment]
@mresti - I should have noticed this before, but as I was creating these stacked charts in Photoshop, it hit me that in the stacked horizontal layout, the way that the quarters appear should be reversed. Right now, Q4 is on the left , with Q1 on the right. In this view, we should see Q1 on the left and Q4 on the right. Is there a way to reverse the order that the data sets are appearing?

File: tests/unit/bar-charts.spec.js (hunk 2 above)
Comment (priley86) [reply tests/unit/bar-charts.spec.js:148]: This looks good to me, but it'd be nice to have at least one unit test for the stacked vertical bar. I wrote one example of this here:
https://github.com/priley86/patternfly/commit/f40e325e78ecc9e2bf9400fcb28ac787ed344899

File: src/js/patternfly-settings.js (hunk 1 above)
Comment (mresti) [reply src/js/patternfly-settings.js:222]: @dlabrecq I send a PR for branch-4.0-dev with this code how you tell me.

File: tests/unit/bar-charts.spec.js (hunk 2 above)
Comment (mresti) [reply tests/unit/bar-charts.spec.js:148]: @priley86 thanks, I update the code using your example. You can review it?

**Comment by mresti** [comment]
Good catch @jennyhaines ! Of course, it can be disabled for stack order of data sets.  By default C3 library for stack order is based on sum of values descendantly. We need add to data config the property `order:null` to disable it.

I updated stacked bar charts config to order data set based on data definition. I give the link for test https://rawgit.com/mresti/patternfly/stacked_bar_chart_dist_review/dist/tests/bar-charts.html

Can @jennyhaines  give me feedback?

**Review by dlabrecq — APPROVED** [review]
LGTM

**Comment by dlabrecq** [comment]
@jennyhaines Do you have any feedback before merging?

**Comment by jennyhaines** [comment]
@dlabrecq @mresti 

It looks great- nice update! It's ready to merge.

**Pull request merged**